### PR TITLE
[9.x] BCC broken on 9.x with AWS SES Mail Transport

### LINKED ISSUE #41357: Blind copies are not sent by sendmail transport
- Laravel Version: 9.3.1
- PHP Version: 8.0.14
- Database Driver & Version: N/A

### Description:
Using a mail controller like this:

```php
class SendmailController extends Controller
{
	public function post_sendmail(Request $request)
	{
		$email = new Sendmail($request->subject, $request->msg);

		Mail::to($request->to)
			->bcc($request->bcc)
			->send($email);

		return "Email sent";
	}
}
```

If MAIL_MAILER=smtp in the .env file, and a suitable SMTP server is configured, the blind copy is sent, but when MAIL_MAILER=sendmail it isn't.

### Steps To Reproduce:
1. Extract the sendmail app from the attached archive to a linux system with a sendmail binary.
2. cd sendmail
3. composer update
4. cd public
5. php -S 127.0.0.1:8080
6. In a web browser, browse to http://127.0.0.1:8080/
7. Complete the fields in the form.
8. Click Send.

[sendmail.tar.gz](https://github.com/laravel/framework/files/8191129/sendmail.tar.gz)

### PR DESCRIPTION
**Pull request by cuppett**
After upgrading to Laravel 9.x, bcc in our application stopped working
with the SES email transport. After reviewing the migration/upgrade
documentation and the code differences from 8.x -> 9.x, it's clear the
change in mailers likely exposed this; however, I also noticed a couple
fundamental, mechanical issues with the implementation:

1) Lack of Destinations specified in the array
2) No adherence to the 50 recipient limit of SES

I've corrected both of those and now can receive To/Cc/Bcc emails.

See Also: https://docs.aws.amazon.com/ses/latest/APIReference/API_SendRawEmail.html
See Also: https://docs.aws.amazon.com/aws-sdk-php/v2/api/class-Aws.Ses.SesClient.html#_sendRawEmail
See Also: https://docs.aws.amazon.com/ses/latest/dg/quotas.html

### CODE REVIEW

File: src/Illuminate/Mail/Transport/SesTransport.php [hunk 1]
```diff
@@ -10,6 +10,12 @@
 
 class SesTransport extends AbstractTransport
 {
+    /**
+     * Amazon SES has a limit on the total number of recipients per message.
+     * See Also: https://docs.aws.amazon.com/ses/latest/dg/quotas.html.
+     */
+    const recipientLimit = 50;
```
Comment (seriquynh) [inline src/Illuminate/Mail/Transport/SesTransport.php:17]: This constant must be RECIPIENT_LIMIT because of coding convention.

Comment (cuppett) [reply src/Illuminate/Mail/Transport/SesTransport.php]: Fixed, thanks!

File: src/Illuminate/Mail/Transport/SesTransport.php [hunk 2]
```diff
@@ -45,21 +51,43 @@ public function __construct(SesClient $ses, $options = [])
...
-                array_merge(
-                    $this->options, [
-                        'Source' => $message->getEnvelope()->getSender()->toString(),
-                        'RawMessage' => [
-                            'Data' => $message->toString(),
-                        ],
-                    ]
-                )
-            );
+            foreach ($this->getRecipients($message) as $recipients) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/SesTransport.php:54]: this looks wrong. we should chunk into batches of 50, instead.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Mail/Transport/SesTransport.php (hunk 2 above)
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/SesTransport.php:54]: your method re-implements chunk. you should just use collection chunk.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Mail/Transport/SesTransport.php [hunk 3]
```diff
@@ -10,6 +10,12 @@
 
 class SesTransport extends AbstractTransport
 {
+    /**
+     * Amazon SES has a limit on the total number of recipients per message.
+     * See Also: https://docs.aws.amazon.com/ses/latest/dg/quotas.html.
+     */
+    const RECIPIENT_LIMIT = 50;
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/SesTransport.php:17]: should be protected

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Mail/Transport/SesTransport.php (hunk 3 above)
Comment (cuppett) [reply src/Illuminate/Mail/Transport/SesTransport.php]: Done. Thanks!

**Comment by cuppett** [comment]
> Does this mean it's also broken in 8.x but hidden? If so we need to target 8.x

Correct. BCC worked, but I didn't see any indication of adherence to the 50 recipient limit.

**Comment by driesvints** [comment]
Please send this to 8.x then, thanks.

**Comment by cuppett** [comment]
I can definitely open another. Won't we still need two PRs? The code in question wasn't exactly identical between the branches.

**Comment by driesvints** [comment]
We'll try to merge this into 9.x

**Comment by cuppett** [comment]
I can tell the PR will be different for 8.x. there are some headers added
for the SES return IDs and a recipient count returned from the function.

Do we want 2 PRs?

**Comment by driesvints** [comment]
@cuppett just send in whatever you think is best 👍

**Comment by cuppett** [comment]
Can do. I may open both just so we have all the right raw material to work with easily.

**Pull request closed without merging**